### PR TITLE
Add `xref` and `yref` to colorbars

### PR DESCRIPTION
**Pull request by hannahker**
**Summary**: Add support for container-referenced positioning for plot colorbars. See demo CodePen here: https://codepen.io/hannahker/pen/eYPMpVV

**API changes**: `yref` and `xref` parameters are added to colorbars.

**Expected behaviour**:
- The colorbar may be positioned relative to the full container area if `colorbar.yref` or `colorbar.xref` are set to `'container'`.
- If a colorbar has container-referenced positioning, then space will be added to the `_fullLayout._reservedMargin` accordingly, meaning that colorbars may be automatically positioned to avoid components such as automargined axes.
     - If `xref='container'`, then margin space will be added to the `_reservedMargin` on the left or the right of a plot area (depending if x > 0.5)
     - If `yref='container'`, then margin space will be added to the `_reservedMargin` on the top or bottom of a plot area (depending if y > 0.5).
     - If both `xref='container'` and `yref='container'`, then margin spacing will be added to the top or bottom if `orientation='v'` and left or right if `orientation='h'`.

The defaults for `yanchor`, `xanchor`, `y`, and `x` have also been adjusted for cases where container-referencing is applied.

**TODO**:
- [x] Constrain `x` and `y` to be within [0, 1] if container referencing is applied. 

See #6589 for similar changes applied for legends.

### CODE REVIEW

**Review by alexcjohnson — APPROVED** [review]
💃 Well done 🎉

File: test/image/baselines/zz-container-colorbar.png
Comment (archmoj) [inline test/image/baselines/zz-container-colorbar.png:1]: This baseline is unused. Correct?
If so, please remove it :pray:

Comment (hannahker) [reply test/image/baselines/zz-container-colorbar.png:1]: Done!

**Comment by archmoj** [comment]
:dancer:

**Pull request merged**